pool.go

```diff
@@ -129,6 +159,18 @@ func (p *Pool) Close() {
 		}
 		client.ClientConn.Close()
 	}
+
+	if unhealthyClients == nil {
+		return
```

Comment:
Note that if `clients` is `nil` but `unhealthyClients` is _not_, we'll never get here. Probably needs some thought, although not sure if this case is possible (presumably an unhealthy client is still a client?)

Reply:
good thought, corrected.